docs/community/contributing.rst

```diff
@@ -0,0 +1,195 @@
+
+**Mocks**
+
+Mocks are similar to stubs but used for **behavioral verification**. For example, a mock can replace a function or an object to check if it is called with correct parameters. A main motivation for using mocks is to avoid complex or time-consuming external dependencies, for example database queries.
+
+- A typical use case of mocks in the pyfar context is hardware communication, for example reading and writing of large files or audio in- and output. These use cases are rare compared to tests performing state verification.
+- In contrast to some other guidelines on mocks, external dependencies do **not** need to be mocked in general. Failing tests due to changes in external packages are meaningful hints to modify the code.
+- Examples of internal mocking can be found in *test_io.py*, indicated by the pytest ``@patch`` calls.
+
+
+Writing the Documentation
+-------------------------
+
+Pyfar follows the `numpy style guide <https://numpydoc.readthedocs.io/en/latest/format.html>`_ for the docstring. A docstring has to consist at least of
+
+- a short and/or extended summary,
+- the Parameters section, and
+- the Returns section.
+
+Optional fields that are often used are
+
+- References,
+- Examples, and
+- Notes.
+
+Here are a few tips to make things run smoothly.
+
+- Use the tags ``:py:func:``, ``:py:mod:``, and ``:py:class:`` to reference functions, modules, and classes from the package itself: For example ``:py:func:`~pyfar.plot.time``` for a link that displays only the function name. For links with custom text use ``:py:mod:`plot functions <pyfar.plot>```.
+- Code snippets and values as well as external modules, classes, functions are marked by double ticks \`\` to appear in mono spaced font, e.g., ``x=3`` or ``pyfar.Signal``.
+- Parameters, returns, and attributes are marked by single ticks \` to appear as emphasized text, e.g., *unit*.
+- Use ``[#]_`` and ``.. [#]`` to get automatically numbered footnotes.
+- Do not use footnotes in the short summary. Only use footnotes in the extended summary if there is a short summary. Otherwise, it messes with the auto-footnotes.
+- If a method or class takes or returns pyfar objects defined in the package for example write ``parameter_name : Signal``. This will create a link to the ``pyfar.Signal`` class.
+- If a method or class takes or returns pyfar objects from other packages for example write ``parameter_name : :py:class:\`~pyfar.classes.audio.Signal\``` to create the link. Note that this requires an intersphinx mapping in `docs/conf.py` in this case ``intersphinx_mapping = {'pyfar': ('https://pyfar.readthedocs.io/en/stable/', None)}``.
+- Plots can be included in the documentation by using the prefix ``.. plot::`` followed by an empty line and an indented block containing the code for the plot. See `pyfar.plot.line.time.py` for examples.
```

Comment:
Aren't there better examples for this in the docs of the used plugin? It's inconvenient to search through code for this.